server/src/main/java/org/apache/lucene/search/uhighlight/CustomUnifiedHighlighter.java

```diff
@@ -200,17 +190,6 @@ protected FieldHighlighter getFieldHighlighter(String field, Query query, Set<Te
             boolean inorder = (mpq.getSlop() == 0);
             return Collections.singletonList(new SpanNearQuery(positionSpanQueries,
                 mpq.getSlop() + positionGaps, inorder));
-        } else if (query instanceof CommonTermsQuery) {
-            CommonTermsQuery ctq = (CommonTermsQuery) query;
-            List<Query> tqs = new ArrayList<> ();
-            for (Term term : ctq.getTerms()) {
-                tqs.add(new TermQuery(term));
-            }
-            return tqs;
-        } else if (query instanceof FunctionScoreQuery) {
-            return Collections.singletonList(((FunctionScoreQuery) query).getSubQuery());
-        } else if (query instanceof ESToParentBlockJoinQuery) {
-            return Collections.singletonList(((ESToParentBlockJoinQuery) query).getChildQuery());
```

Comment:
Do we not need these any more?

Reply:
The unified highlighter uses a QueryVisitor to extract the terms now so I added a `visit` implementation for these queries since this code is only called to extract phrase and span queries:
https://github.com/elastic/elasticsearch/pull/40952/files#diff-4d092acdc70f570b9d18594dcffb95a6R80
https://github.com/elastic/elasticsearch/pull/40952/files#diff-80c91092d36bb5d49bdd3ef462f75e40R201

Reply:
👍